include/libopencm3/stm32/h7/pwr.h

```diff
@@ -98,27 +196,49 @@ enum pwr_svos_scale {
 };
 
 enum pwr_vos_scale {
-  PWR_VOS_SCALE_0 = 0,    /* Note: This state requires SYSCFG ODEN set. */
-  PWR_VOS_SCALE_1 = (PWR_D3CR_VOS_SCALE_1 << PWR_D3CR_VOS_SHIFT),
-  PWR_VOS_SCALE_2 = (PWR_D3CR_VOS_SCALE_2 << PWR_D3CR_VOS_SHIFT),
-  PWR_VOS_SCALE_3 = (PWR_D3CR_VOS_SCALE_3 << PWR_D3CR_VOS_SHIFT)
+  PWR_VOS_SCALE_UNDEFINED = 0,
+  PWR_VOS_SCALE_0,
+  PWR_VOS_SCALE_1,
+  PWR_VOS_SCALE_2,
+  PWR_VOS_SCALE_3,
 };
 
 BEGIN_DECLS
 
 /** @defgroup pwr_peripheral_api PWR Peripheral API
  * @ingroup peripheral_apis
 @{*/
-
 /** Set power subsystem to utilize the LDO for CPU. */
 void pwr_set_mode_ldo(void);
 
-/** Set the voltage scaling/strength for the internal LDO when in Stop mode.
+/** Specific STM32H742/43/50/53 LDO mode setting.. */
+void pwr_set_mode_scu_ldo(void);
+
+/** Set power subsystem to utilize the SMPS run through the LDO for CPU.
+ *  @param[in] supply_external  Supply is powering external circuits, enable high power mode.
+ *  @param[in] smps_level  Voltage level from available PWR_CR3_SMPSLEVEL_XXX settings (1.8V/2.5V)
+ *  @param[in] use_ldo  Set this value to true if the internal LDO should be enabled.
+ */
+void pwr_set_mode_smps_ldo(bool supply_external, uint32_t smps_level, bool use_ldo);
+
+/** Set power system based on "System Supply Configurations" table in reference manual.
+ *  @param[in] mode  Mode mapping to a mode in the system configuration. Note special SCU modes.
+ *  @param[in] smps_level  Optional, only needed if using an EXT_SMPS or SMPS_LDO mode.
```

Comment:
What is "optional" here, are you mean to provide zero?  is this from PWR_CR3_SMPSLEVEL_xxxx as well? can I add doc groups for them?  (Don't worry, I'm merging this, it looks excelltne thanks, just want to tidy up the docs as much as I can while it's fresh)

Reply:
See https://github.com/libopencm3/libopencm3/pull/1307 for how I think I've understood this.